staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go

```diff
@@ -25,67 +25,9 @@ import (
 	"k8s.io/apimachinery/pkg/conversion"
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-func AddConversionFuncs(scheme *runtime.Scheme) error {
-	return scheme.AddConversionFuncs(
```

Comment:
This doesn't work now.

The reason is that we currently have two different paths for:
[1] scheme.AddConversionFuncs [used for manual conversions]
[2] scheme.AddConversionFunc [used for registered via autogenerate stuff]

In particular, as soon as you enter the "defaultConversion" path, the ones that are registered via autogenerated code [2], are no longer available for you, because this code only is checking the first ones:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/conversion/converter.go#L564

So it seems we need to clean the converter code and all places that still rely on it before we will be able to do that.

@smarterclayton @liggitt @deads2k - FYI ^^

Reply:
I opened https://github.com/kubernetes/kubernetes/issues/84452 for this.